### PR TITLE
Option to enable cors for editor

### PR DESCRIPTION
**Pull request by Trisooma**
I want to be able to have my own editor talk to the node-red server - to be able to do this CORS needs to be enabled for the server. I'm really not sure if this is the best way to do it, but it works for me..

Comments welcome

### CODE REVIEW

**Comment by knolleary** [comment]
I can certainly see the requirement to enable CORS handling on the admin routes - but I don't think your implementation is the right way to do it. These options will apply CORS to both admin and http routes - not just admin routes.

If we're going to add this, it ought to be `httpAdminCors` and only get applied to the admin route in `red/api/index.js`

**Comment by Trisooma** [comment]
Thanks for pointing me in the right direction! I renamed the setting to `httpAdminCors` and it only gets applied on the admin routes.

File: red.js [hunk 1]
```diff
@@ -267,6 +267,9 @@ RED.start().then(function() {
             }
             process.title = 'node-red';
             RED.log.info(RED.log._("server.now-running", {listenpath:getListenPath()}));
+            if (settings.httpAdminCors !== undefined) {
+                RED.log.info(RED.log._("server.admin-cors-enabled"));
+            }
         });
```
Comment (knolleary) [inline red.js]: We don't log anything else like this - I don't see a need to here.

File: red/api/index.js [hunk 2]
```diff
@@ -112,6 +112,21 @@ function init(_server,_runtime) {
...
+                if(settings.httpAdminCors.methods) {
+                    res.header('Access-Control-Allow-Methods', settings.httpAdminCors.methods);
+                }
+                if(settings.httpAdminCors.headers) {
+                    res.header('Access-Control-Allow-Headers', settings.httpAdminCors.headers);
+                }
+                next();
+            };
+            adminApp.use(allowCrossDomain);
+        }
```
Comment (knolleary) [inline red/api/index.js]: We already pre-req the `cors` module for doing this. Is there any reason not to reuse that here as we do [here](https://github.com/node-red/node-red/blob/master/nodes/core/io/21-httpin.js#L162)?

**Comment by Trisooma** [comment]
Since I don't user node very often I'm not sure what you mean by reusing. I used a similar construct as the file you pointed me too. This way (adding handler globally) avoid having to specify it on every route.

**Comment by knolleary** [comment]
:+1:

**Pull request merged**